### PR TITLE
[backport] Support `allow_pickle` in `cupy.load` and `cupy.save`

### PR DESCRIPTION
**Pull request by asi1024**
Backport of #2290

### CODE REVIEW

**Comment by asi1024** [comment]
[automatic post] Jenkins, test this please.

**Comment by kmaehashi** [comment]
Could you fix the default to `True` to avoid `no-compat`?
https://github.com/cupy/cupy/pull/2290#issuecomment-508626335

File: cupy/io/npz.py [hunk 1]
```diff
@@ -35,6 +40,16 @@ def load(file, mmap_mode=None):
         mmap_mode (None, 'r+', 'r', 'w+', 'c'): If not ``None``, memory-map the
             file to construct an intermediate :class:`numpy.ndarray` object and
             transfer it to the current device.
+        allow_pickle (bool): Allow loading pickled object arrays stored in npy
+            files. Reasons for disallowing pickles include security, as
+            loading pickled data can execute arbitrary code. If pickles are
+            disallowed, loading object arrays will fail.
+            Please be aware that CuPy does not support arrays with dtype of
+            `object`.
+            The default is False.
```
Comment (kmaehashi) [inline cupy/io/npz.py]:
```suggestion
            The default is True.
```

File: cupy/io/npz.py [hunk 2]
```diff
@@ -44,7 +59,14 @@ def load(file, mmap_mode=None):
     .. seealso:: :func:`numpy.load`
 
     """
-    obj = numpy.load(file, mmap_mode)
+    if _support_allow_pickle:
+        allow_pickle = False if allow_pickle is None else allow_pickle
```
Comment (kmaehashi) [inline cupy/io/npz.py]:
```suggestion
        allow_pickle = True if allow_pickle is None else allow_pickle
```

**Comment by asi1024** [comment]
@kmaehashi Could you take a look?

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**